src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java

```diff
@@ -83,6 +82,41 @@ public class PaymentGatewayClient {
 
     private final Map<String, URI> checkoutRedirectBeApiCallUriMap;
 
+    static final Map<RedirectPaymentMethodId, String> redirectMethodsDescriptions = Map.of(
+            RedirectPaymentMethodId.RBPR,
+            "Poste addebito in conto Retail",
+            RedirectPaymentMethodId.RBPB,
+            "Poste addebito in conto Business",
+            RedirectPaymentMethodId.RBPP,
+            "Paga con BottonePostePay",
+            RedirectPaymentMethodId.RPIC,
+            "Pago in Conto Intesa",
+            RedirectPaymentMethodId.RBPS,
+            "SCRIGNO Internet Banking"
+    );
+
+    public enum RedirectPaymentMethodId {
+        RBPR,
+        RBPB,
+        RBPP,
+        RPIC,
+        RBPS;
+
+        private static final Map<String, RedirectPaymentMethodId> lookupMap = Arrays
+                .stream(RedirectPaymentMethodId.values())
+                .collect(Collectors.toMap(Enum::toString, Function.identity()));
+
+        static RedirectPaymentMethodId fromPaymentTypeCode(String paymentTypeCode) {
+            RedirectPaymentMethodId converted = lookupMap.get(paymentTypeCode);
+            if (converted == null) {
+                throw new InvalidRequestException(
+                        "Unmanaged payment method with type code: [%s]".formatted(paymentTypeCode)
+                );
+            }
+            return converted;
+        }
+    }
```

Comment:
Isn't this the same as `valueOf()`? (except for the exception thrown)

Reply:
Yes, I put it just to thrown the Invalid request exception to return a 400 bad request instead of an Internal Server Error.